state/txindex/indexer.go

```diff
@@ -6,28 +6,27 @@ import (
 	"github.com/tendermint/tendermint/types"
 )
 
-// Indexer interface defines methods to index and search transactions.
+// TxIndexer interface defines methods to index and search transactions.
+// TODO: Add the ability to create an independent TxIndexer.
```

Comment:
what's this TODO about ?

Reply:
In `func (s *State) Copy() *State` we copy everything except the underlying TxIndexer. It introduces a special case into state, where you can get a completely independent state except for the TxIndexer.

Is it okay if the TxIndexer is never copied and always points to the same underlying one or should we add `Copy()` to the TxIndexer interface to allow for independent ones?

Reply:
I think its ok